### PR TITLE
Retries creating workspace content dir in ShelvingService.

### LINKED ISSUE #4952: File exists @ dir_s_mkdir - ShelvingService
## Backtrace

    line 30 of [PROJECT_ROOT]/app/services/shelving_service.rb: shelve
    line 9 of [PROJECT_ROOT]/app/services/shelving_service.rb: shelve
    line 15 of [PROJECT_ROOT]/app/jobs/shelve_job.rb: perform

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/50568/faults/94972560)

### PR DESCRIPTION
**Pull request by justinlittman**
closes #4952

## Why was this change made? 🤔
Less noisy integration tests.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Integration

### CODE REVIEW

**Review by jcoyne — CHANGES_REQUESTED** [review]
I believe this will make the problem worse.

Currently it fails because another process has already created the directory.   If you keep retrying, you will continue to find that the directory exists and still cannot be created.

**Comment by justinlittman** [comment]
Here's the relevant code: https://github.com/sul-dlss/druid-tools/blob/main/lib/druid_tools/druid.rb#L88-L100

There are 2 checks to see if the directory exists prior to create (the first one should skip the create; the second one should raise). I'm guessing that there is probably a file system sync problem that is causing both of those checks to report that the directory doesn't exist. I'd expect that if retried the file system sync will resolve and the code will work as expected.

**Comment by jcoyne** [comment]
@justinlittman If you have two processes running simultaneously and they each ask "Directory x created yet?" they will each get a "no". Then they will both try to create it.  One will succeed and the other will get this error.  A retry will just give the error again.

**Comment by justinlittman** [comment]
On the second process's retry, it will discover that the directory exists (https://github.com/sul-dlss/druid-tools/blob/main/lib/druid_tools/druid.rb#L90) and skip the mkdir.

File: app/services/shelving_service.rb [hunk 1]
```diff
@@ -27,7 +27,14 @@ def shelve
     preserve_diff = content_diff('preserve')
     shelve_diff = content_diff('shelve')
 
-    workspace_content_pathname = Pathname(workspace_druid.content_dir(true))
+    # This addresses a race condition.
```
Comment (jcoyne) [inline app/services/shelving_service.rb:30]: can you be more specific about the race condition and why the retry resolves it in this comment?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/services/shelving_service.rb (hunk 1 above)
Comment (jcoyne) [reply app/services/shelving_service.rb:30]: Should the retry be upstream in the gem? Don't we want it fixed for all users?

Comment (justinlittman) [reply app/services/shelving_service.rb:30]: I'm reluctant to fix in the gem, as the fix might have unintended consequences on other file systems.

Adding more detail to the comment.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**